docs/framework/architecture/ui-components.md

```diff
@@ -470,6 +470,205 @@ disabledDropdown.render();
 document.getElementById( 'dropdown-disabled' ).append( disabledDropdown.element );
 ```
 
+## Dialog
+
+{@snippet framework/ui/ui-dialog}
+
+A dialog window is a draggable popup that can be displayed on top of the editor contents and remains open while user interacts with the editing area. You can use it to display any detached UI. You can create a dialog as an instance of the {@link module:ui/dialog/dialogview~DialogView `DialogView`} class.
+
+```js
+import { Locale } from '@ckeditor/ckeditor5-utils';
+import { ButtonView, DialogView, View } from '@ckeditor/ckeditor5-ui';
+
+const locale = new Locale();
+
+const dialogButton = new ButtonView();
+dialogButton.set( {
+	label: 'Show a dialog',
+	withText: true,
+	class: 'ck-button-action'
+} );
+dialogButton.render();
+
+let dialog;
+
+dialogButton.on( 'execute', () => {
+	if ( dialogButton.isOn ) {
+		hideDialog( dialog );
+
+		return;
+	}
+
+	dialogButton.isOn = true;
+
+	dialog = new DialogView( locale, {
```

Comment:
I'm not sure if this whole snippet makes any sense. `DialogView` has never meant to be used without the `Dialog` plugin. It's quite obvious it's missing specific editor couplings such as viewport offset or DOM roots. We never meant developers to use dialogs this way.

The examples (snippets) presented in this guide should be copy-pasteable pieces of code developers can use in their integrations. What you presented here is more or less how to build your own dialogs system using some pieces of our dialogs system :D Not saying it's impossible, but this is not the way go.

Either we create an editor and showcase how to use the dialog plugin properly here or I say we don't do anything in this guide. Notice how silent it is on the contextual balloon plugin (that also makes no sense without an editor). `BalloonPanelView`, however, unlike `DialogView`, can be used standalone and does not need contextual balloon plugin to be useful (that's why it got a section here).